code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1146,102 +1152,132 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 		return "", common.NewError("allocation_updating_failed",
 			"can't update expired allocation")
 	}
-	// update allocation transaction hash
-	alloc.Tx = t.Hash
-	if len(request.Name) > 0 {
-		alloc.Name = request.Name
-	}
-
-	// adjust expiration
-	var newExpiration = alloc.Expiration + request.Expiration
-	// close allocation now
-
-	if newExpiration <= t.CreationDate {
-		return sc.closeAllocation(t, alloc, balances) // update alloc tx, expir
-	}
-
-	// an allocation can't be shorter than configured in SC
-	// (prevent allocation shortening for entire period)
-	if newExpiration-t.CreationDate < toSeconds(conf.MinAllocDuration) {
-		return "", common.NewError("allocation_updating_failed",
-			"allocation duration becomes too short")
-	}
-
-	var newSize = request.Size + alloc.Size
-	if newSize < conf.MinAllocSize || newSize < alloc.UsedSize {
-		return "", common.NewError("allocation_updating_failed",
-			"allocation size becomes too small")
-	}
 
-	// get blobber of the allocation to update them
 	var blobbers []*StorageNode
 	if blobbers, err = sc.getAllocationBlobbers(alloc, balances); err != nil {
 		return "", common.NewError("allocation_updating_failed",
 			err.Error())
 	}
 
-	if len(request.AddBlobberId) > 0 {
-		blobbers, err = alloc.changeBlobbers(
-			conf, blobbers, request.AddBlobberId, request.RemoveBlobberId, sc, t.CreationDate, balances,
-		)
-		if err != nil {
-			return "", common.NewError("allocation_updating_failed", err.Error())
-		}
-	}
 
-	if len(blobbers) != len(alloc.BlobberAllocs) {
-		return "", common.NewError("allocation_updating_failed",
-			"error allocation blobber size mismatch")
-	}
+	// If the txn client_id is not the owner of the allocation, should just be able to extend the allocation if permissible
+	// This way, even if an atttacker of an innocent user incorrectly tries to modify any other part of the allocation, it will not have any effect
+	if t.ClientID != alloc.Owner /* Third-party actions */ {
```

Comment:
is it possible to extend allocation without changing expiration?

Reply:
Afaik (and cmiiw) extending the allocation means either increasing size or extending expiration time. So yes the user (third-party) can either increase size alone, expiry alone or both. Are we on the same page regarding the meaning of extending the allocation ?